services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -109,9 +111,28 @@ class HakemusService(
                     hanke = hanke))
         val hakemus = entity.toHakemus()
         hakemusLoggingService.logCreate(hakemus, userId)
+        // A hanke with a hakemus should be in RAKENTAMINEN phase
+        updateHankevaiheToRakentaminen(hanke, userId)
         return hakemus
     }
 
+    /** Update the hankevaihe of a Hanke to RAKENTAMINEN if it's not already there. */
+    private fun updateHankevaiheToRakentaminen(hanke: HankeEntity, userId: String) {
+        if (hanke.vaihe != Hankevaihe.RAKENTAMINEN) {
+            val geometriatMap = hankealueService.geometryMapFrom(hanke.alueet)
+            val hankeBeforeUpdate = HankeMapper.domainFrom(hanke, geometriatMap)
+            hanke.vaihe = Hankevaihe.RAKENTAMINEN
+            hanke.version = hanke.version?.inc() ?: 1
+            hanke.modifiedByUserId = userId
+            hanke.modifiedAt = getCurrentTimeUTCAsLocalTime()
+            hanke.generated = false
+
+            hankeRepository.save(hanke)
```

Comment:
Saving the hanke can change it, so the return value should be used for the logging.

Reply:
You're right. Changed it.